### PR TITLE
Fix Objective-C test pods for Cronet tests

### PR DESCRIPTION
**Pull request by muxi**
Fixed by compiling gRPC with preprocessor macro GRPC_COMPILE_WITH_CRONET

### CODE REVIEW

File: gRPC-Core.podspec [hunk 1]
```diff
@@ -770,12 +770,16 @@ Pod::Spec.new do |s|
     ss.source_files = 'include/grpc/grpc_cronet.h'
   end
 
+  s.subspec 'Cronet-Implement' do |ss|
```
Comment (jcanizales) [inline gRPC-Core.podspec]: nit: Implementation

File: gRPC-Core.podspec [hunk 2]
```diff
@@ -770,12 +770,16 @@ Pod::Spec.new do |s|
...
+    ss.header_mappings_dir = '.'
+    ss.source_files = 'src/core/ext/transport/cronet/client/secure/cronet_channel_create.c',
+                      'src/core/ext/transport/cronet/transport/cronet_transport.c'
+  end
+
   s.subspec 'Cronet-Tests' do |ss|
```
Comment (jcanizales) [inline gRPC-Core.podspec]: Those aren't Cronet-specific anymore, so maybe just `'Tests'`?

**Comment by jcanizales** [comment]
Travis is unhappy by:

```
Undefined symbols for architecture i386:
  "_grpc_cronet_secure_channel_create", referenced from:
      -[GRPCChannel initWithHost:cronetEngine:channelArgs:] in libgRPC.a(GRPCChannel.o)
```

We're missing something :)

File: src/objective-c/tests/Podfile [hunk 3]
```diff
@@ -91,13 +82,23 @@ post_install do |installer|
     target.build_configurations.each do |config|
       config.build_settings['GCC_TREAT_WARNINGS_AS_ERRORS'] = 'YES'
     end
-    if target.name == 'gRPC-Core' or target.name == 'gRPC-Core.default-Cronet-Interface-Cronet-Tests'
+    if target.name == 'gRPC-Core' or target.name == 'gRPC-Core.default-Cronet-Implementation-Cronet-Interface-Tests' or target.name == 'gRPC-Core.default-Cronet-Implementation'
```
Comment (jcanizales) [inline src/objective-c/tests/Podfile]: nit: So we don't have to change it ever again:
```ruby
# Target names generated by Cocoapods are either of the form 'gRPC-Core' or
# 'gRPC-Core.default-' + <optional subspecs separated by dashes>
if target.name.start_with?('gRPC-Core')
```

Comment (muxi) [inline src/objective-c/tests/Podfile]: Ahh nice Ruby features :) I forgot it's Ruby already.

Comment (jcanizales) [inline src/objective-c/tests/Podfile]: Including the comment will prevent people from going: "Huh? Do we have more than one pod that starts with gRPC-Core?". Plus it'll help debug things if Cocoapods changes that naming in the future, by explaining what our intention was.

**Comment by muxi** [comment]
test this please

**Comment by jcanizales** [comment]
The interop test failure can't be related to this PR. The Python docker image is failing to build for some reason.

**Comment by pgrosu** [comment]
@jcanizales Just set the locale environmental variables like this:

`export LANG=en_US.UTF-8` 

in either  `grpc/tools/run_tests/build_python.sh` or before you run the script, and see if that fixes it.

**Comment by jcanizales** [comment]
Please test this.

**Comment by adelez** [comment]
@pgrosu, I'm now having problem building Python tests. See #7888. Can you please take a look at it? Thanks!

**Pull request merged**